src/include/simeng/OS/SyscallHandler.hh

```diff
@@ -103,13 +105,106 @@ struct linux_dirent64 {
   char* d_name;       // Filename (null-terminated)
 };
 
-/** A Linux kernel syscall emulation implementation, which mimics the
-   responses to Linux system calls. */
+/** The types of changes that can be made to values within the process state. */
+enum class ChangeType { REPLACEMENT, INCREMENT, DECREMENT };
+
+/** A structure describing a set of changes to the process state. */
+struct ProcessStateChange {
+  /** Type of changes to be made */
+  ChangeType type;
+  /** Registers to modify */
+  std::vector<Register> modifiedRegisters;
+  /** Values to set modified registers to */
+  std::vector<RegisterValue> modifiedRegisterValues;
+  /** Memory address/width pairs to modify */
+  std::vector<MemoryAccessTarget> memoryAddresses;
+  /** Values to write to memory */
+  std::vector<RegisterValue> memoryAddressValues;
+};
+
+/** The result from a handled syscall. */
+struct SyscallResult {
+  /** Whether execution should halt. */
```

Comment:
please add space between variable names and comments and add `_` to all member variables

Reply:
Spaces added, `_` omitted with reasoning similar to https://github.com/UoB-HPC/SimEng/pull/285#discussion_r1114109519